src/Chiron/Chiron.fs

```diff
@@ -389,793 +662,1342 @@ module internal Escaping =
+                    |> JsonResult.applyDelayedWithPolicy StopOnFirstError decodeC c
+                    |> JsonResult.applyDelayedWithPolicy StopOnFirstError decodeD d
+                    |> JsonResult.applyDelayedWithPolicy StopOnFirstError decodeE e
+                    |> JsonResult.applyDelayedWithPolicy StopOnFirstError decodeF f
+                    |> JsonResult.applyDelayedWithPolicy StopOnFirstError decodeG g
+
+            let inline jsonTuple7ToTuple7 mkTuple (decodeA: Decoder<Json,'a>) (decodeB: Decoder<Json,'b>) (decodeC: Decoder<Json,'c>) (decodeD: Decoder<Json,'d>) (decodeE: Decoder<Json,'e>) (decodeF: Decoder<Json,'f>) (decodeG: Decoder<Json,'g>) =
+                fun (a, b, c, d, e, f, g) ->
+                    JsonResult.map mkTuple (decodeA a)
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeB b
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeC c
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeD d
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeE e
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeF f
+                    |> JsonResult.applyDelayedWithPolicy ContinueOnError decodeG g
+
+            let tuple7WithQuick (decodeA: Decoder<Json,'a>) (decodeB: Decoder<Json,'b>) (decodeC: Decoder<Json,'c>) (decodeD: Decoder<Json,'d>) (decodeE: Decoder<Json,'e>) (decodeF: Decoder<Json,'f>) (decodeG: Decoder<Json,'g>) =
+                tuple7 >=> tuple7WithTaggedDecoders jsonTuple7ToTuple7Quick decodeA decodeB decodeC decodeD decodeE decodeF decodeG
+
+            let tuple7With (decodeA: Decoder<Json,'a>) (decodeB: Decoder<Json,'b>) (decodeC: Decoder<Json,'c>) (decodeD: Decoder<Json,'d>) (decodeE: Decoder<Json,'e>) (decodeF: Decoder<Json,'f>) (decodeG: Decoder<Json,'g>) =
+                tuple7 >=> tuple7WithTaggedDecoders jsonTuple7ToTuple7 decodeA decodeB decodeC decodeD decodeE decodeF decodeG
+
+            let number =
+                do ()
+                function
+                | Json.Number n -> JsonResult.pass n
+                | json -> JsonResult.typeMismatch JsonMemberType.Number json
+
+            let int16 =
+                number >=> Decoder.fromThrowingConverter (fun s -> System.Int16.Parse(s, CultureInfo.InvariantCulture))
+
+            let int =
+                number >=> Decoder.fromThrowingConverter (fun s -> System.Int32.Parse(s, CultureInfo.InvariantCulture))
+
+            let int64 =
+                number >=> Decoder.fromThrowingConverter (fun s -> System.Int64.Parse(s, CultureInfo.InvariantCulture))
+
+            let uint16 =
+                number >=> Decoder.fromThrowingConverter (fun s -> System.UInt16.Parse(s, CultureInfo.InvariantCulture))
+
+            let uint32 =
+                number >=> Decoder.fromThrowingConverter (fun s -> System.UInt32.P
```

Comment:
I don't pretend to understand everything going on this library, but I am curious about this: why use SRTPs here?